playwright/_impl/_set_input_files_helpers.py

```diff
@@ -0,0 +1,80 @@
+import base64
+import os
+import shutil
+from pathlib import Path
+from posixpath import basename
+from typing import TYPE_CHECKING, List, Union
+
+from playwright._impl._helper import Error, async_readfile
+from playwright._impl._writable_stream import WritableStream
+
+if TYPE_CHECKING:  # pragma: no cover
+    from playwright._impl._browser_context import BrowserContext
+
+from playwright._impl._api_structures import FilePayload
+
+SIZE_LIMIT_IN_BYTES = 50 * 1024 * 1024
+
+
+async def convert_input_files(
+    files: Union[str, Path, FilePayload, List[Union[str, Path]], List[FilePayload]],
+    context: "BrowserContext",
+):
+    files = files if isinstance(files, list) else [files]
+
+    has_large_buffer = list(
+        filter(
+            lambda f: not isinstance(f, (str, Path))
+            and len(f.get("buffer", "")) > SIZE_LIMIT_IN_BYTES,
+            files,
+        )
+    )
+    if has_large_buffer:
+        raise Error(
+            "Cannot set buffer larger than 50Mb, please write it to a file and pass its path instead."
+        )
+
+    has_large_file = list(
```

Comment:
seems like this does not check the upper file limits, lets do something like this instead?

`any([os.stat(f).st_size > SIZE_LIMIT_IN_BYTES for f in files if isinstance(f, (str, Path))])`